File: practice/algorithms/implementation/beautiful_triplets/beautiful_triplets.cpp
#include <vector>
#include <iostream>

using namespace std;

int main()
{
    int n, d, res = 0;
    cin >> n >> d;
    vector<int> arr(n);
    for (int pos = 0; pos < n; ++pos)
    {
        cin >> arr[pos];
    }
    
    for (int pos1 = 0; pos1 < n; ++pos1)
    {
        for (int pos2 = pos1 + 1; pos2 < n; ++pos2)
        {
            if (arr[pos2] > arr[pos1] + d)
            {
                break;
            }
            if (arr[pos2] == arr[pos1] + d)
            {
                for (int pos3 = pos2 + 1; pos3 < n; ++pos3)
                {
                    if (arr[pos3] > arr[pos2] + d)
                    {
                        break;
                    }
                    if (arr[pos3] == arr[pos2] + d)
                    {
                        res++;
                    }
                }
            }    
        }
    }
    
    cout << res << endl;
    
    return 0;
}

File: practice/algorithms/implementation/beautiful_triplets/beautiful_triplets.py
#!/usr/bin/env python

n, d = map(int, raw_input().strip().split())
arr = map(int, raw_input().strip().split())

res = 0
for POS1 in xrange(n):
    for POS2 in xrange(POS1 + 1, n):
        if arr[POS2] - arr[POS1] > d:
            break
        if arr[POS2] - arr[POS1] == d:
            for POS3 in xrange(POS2 + 1, n):
                if arr[POS3] - arr[POS2] > d:
                    break
                if arr[POS3] - arr[POS2] == d:
                    res += 1
print res

